kiwi/system/setup.py

```diff
@@ -239,6 +239,38 @@ def setup_machine_id(self):
         with open(machine_id, 'w'):
             pass
 
+    def setup_permissions(self):
+        """
+        Check and Fix permissions using chkstat
+
+        Call chkstat in system mode which reads /etc/sysconfig/security
+        to determine the configured security level and applies the
+        appropriate permission definitions from the /etc/permissions*
+        files. It's possible to provide those files as overlay files
+        in the image description to apply a certain permission setup
+        when needed. Otherwise the default setup as provided on the
+        package level applies.
+
+        It's required that the image root system has chkstat installed.
+        If not present KIWI skips this step and continuous with a
+        warning.
+        """
+        chkstat_search_env = {
+            'PATH': os.sep.join([self.root_dir, 'usr', 'bin'])
+        }
+        chkstat = Path.which(
```

Comment:
Just in case you missed it, this code could benefit from the KiwiCommandNotFound exception logic, it is raised exactly with this same path check. So this could be coded as:

```python
try:
  log.info('Check/Fix File Permissions')
  Command.run(
    ['chroot', self.root_dir, 'chkstat', '--system', '--set']
  )
except KiwiCommandNotFound:
  log.warning(
    'chkstat not found in image. File Permissions Check skipped'
  )
```

Anyway I am fine with the current implementation.

Reply:
This is not the same. KiwiCommandNotFound is thrown if the command which is expected to be the first element in the list, is not found. In this case this applies to the ```chroot``` command. But my code explicitly tests if the ```chkstat``` command exists in the image root dir.

chkstat supports the --root flag but that would call chkstat from the host and I explicitly wanted to call the program from the image root because I don't trust the host installed chkstat to do the right thing if it's not from the same distro as the image

Reply:
You are absolutely right sorry, I answered without noticing it was a chroot command.